test/4_integration.js

```diff
@@ -1,108 +1,242 @@
-		storage().clear();
-		var branch = new Branch();
+	var sampleParams = {
+		tags: [ 'tag1', 'tag2' ],
+		channel: 'sample app',
+		feature: 'create link',
+		stage: 'created link',
+		type: 1,
+		data: {
+			mydata: 'bar',
+			'$desktop_url': 'https://cdn.branch.io/example.html',
+			'$og_title': 'Branch Metrics',
+			'$og_description': 'Branch Metrics',
+			'$og_image_url': 'http://branch.io/img/logo_icon_white.png'
+		}
+	};
+
+	var branchInit = function(init, assert) {
+		if (init) { branch.init(browser_fingerprint_id); }
+		if (assert) { assert.equal(requests.length, 1); }
+		requests[0].callback(browser_fingerprint_id);
+		if (assert) { assert.equal(requests.length, 2); }
+		requests[1].respond(200,
+				{ "Content-Type": "application/json" },
+				'{ "session_id":"123088518049178533", "identity_id":"114720603218387056", "device_fingerprint_id":null, "browser_fingerprint_id":"79336952217731267", "link":"https://bnc.lt/i/4LYQTXE0_k", "identity":"Branch","has_app":true }');
+	};
+
+	describe('init', function() {
+		it('should call api with params and version', function(done) {
+			var assert = testUtils.plan(4, done);
+			branch.init(browser_fingerprint_id, function(err, data) {
+				assert.deepEqual(data,
+					{
+						data: null,
+						referring_identity: null,
+						identity: "Branch",
+						has_app: true
+					},
+					'Expected response returned');
+			});
+			branchInit(false, assert);
+			assert.equal(requests[0].src, 'https://bnc.lt/_r?v=' + config.version + '&callback=branch_callback__' + 0, 'Endpoint correct');
+		});
+
+		it('should support being called without a callback', function(done) {
+			var assert = testUtils.plan(3, done);
+			branch.init(browser_fingerprint_id);
+			branchInit(false, assert);
+			assert.equal(requests[0].src, 'https://bnc.lt/_r?v=' + config.version + '&callback=branch_callback__' + 1, 'Endpoint correct');
+		});
 
-		sandbox.stub(branch._server, "request", function(resource, obj, storage, callback) {
-			requests.push({
-				resource: resource,
-				obj: obj,
-				callback: callback
+		it('should return error to callback', function(done) {
+			var assert = testUtils.plan(1, done);
+			branch.init(browser_fingerprint_id, function(err) {
+				 assert.equal(err.message, 'Error in API: 400');
 			});
+			requests[0].callback(browser_fingerprint_id);
+			requests[1].respond(400);
 		});
 
-		if (runInit) {
-			branch.init(branch_sample_key);
-			requests[0].callback(null, browser_fingerprint_id);
-			requests[1].callback(null, { session_id: session_id, browser_fingerprint_id: browser_fingerprint_id, identity_id: identity_id });
-			requests = [];
-		}
+		it('should store in session and call open with link_identifier from hash', function(done) {
+			var assert = testUtils.plan(1, done);
+			testUtils.go("#r:12345");
+			branch.init(branch_sample_key, function(err, data) {
+				assert.equal(utils.readStore(branch._storage).click_id, '12345', 'click_id from link_identifier hash stored in session_id');
```

Comment:
ahh ok cool. hmm... it's like a weird coupling of unit and integration testing. let's find other ways to check this -- for example, do a sendSMS and check the url it gets sent to